odoo/model.go

```diff
@@ -0,0 +1,92 @@
+package odoo
+
+import "fmt"
+
+// SearchReadModel is used as "params" in requests to "dataset/search_read" endpoints.
+type SearchReadModel struct {
+	Model  string   `json:"model,omitempty"`
+	Domain []Filter `json:"domain,omitempty"`
+	Fields []string `json:"fields,omitempty"`
+	Limit  int      `json:"limit,omitempty"`
+	Offset int      `json:"offset,omitempty"`
+}
+
+// Filter to use in queries, usually in the format of
+// [predicate, operator, value], eg ["employee_id.user_id.id", "=", 123]
+type Filter interface{}
```

Comment:
I'd ensure `Filter` can only be created by this package by adding a private method or just use `interface{}`. I see no value in this type like this.

Reply:
It adds a bit more semantics and docs than just `interface{}` :shrug: